MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionPresenter.cpp

```diff
@@ -1789,8 +1797,29 @@ void EnggDiffractionPresenter::loadVanadiumPrecalcWorkspaces(
   // algCurves->getProperty("OutputWorkspace");
   vanCurvesWS = ADS.retrieveWS<MatrixWorkspace>(curvesWSName);
 
-  saveOpenGenie(curvesWSName, "1-1200", "North", vanNo);
-  saveOpenGenie(curvesWSName, "1201-2400", "South", vanNo);
+  if (specNos != "") {
+    if (specNos == "North") {
+      // when north bank is selected while cropped calib
+      saveOpenGenie(curvesWSName, "1-1200", "North", vanNo);
+    } else if (specNos == "South") {
+      // when south bank is selected while cropped calib
+      saveOpenGenie(curvesWSName, "1201-2400", "South", vanNo);
+    } else {
+
+      // when spectrumIDs are provided
+      std::string CustomisedBankName = m_view->currentCalibCustomisedBankName();
```

Comment:
Code changes look very good. I only spotted this very minor thing: the string in this line could be const (not worth changing).